src/utility.jl

```diff
@@ -74,3 +75,260 @@ function visualize_flow(method::ColorBased, flow::Array{SVector{2, Float64}, 2};
+
+bytes contain -
+  0-3     tag: "PIEH" in ASCII, which in little endian happens to be the float 202021.25
+          (just a sanity check that floats are represented correctly)
+  4-7     width as an integer
+  8-11    height as an integer
+  12-end  data (width*height*2*4 bytes total)
+          the float values for u and v, interleaved, in row order, i.e.,
+          u[row0,col0], v[row0,col0], u[row0,col1], v[row0,col1], ...
+
+# Arguements
+The file_path parameter needs to be a string, storing the address of .flo file.
+
+# Example
+
+Read flow file from .flo file and visualize the flow in HSV
+```julia
+
+using ImageTracking
+
+file_path = "./Example_flow.flo"
+flow = read_flow_file(file_path)
+result = visualize_flow(flow, ColorBased(), RasterConvention())
+imshow(RGB.(result))
+```
+
+# References
+[1] http://vision.middlebury.edu/flow/data/
+"""
+# first four bytes to be same in little endian
+# check this when reading a .flo file
+FLOAT_TAG = 202021.25
+function read_flow_file(file_path::String)
+    
+    io = open(file_path, "r")
+    seekstart(io)
+    
+    tag = read(io, Float32) # reading the initial tag 
+    
+    if tag != FLOAT_TAG
+        print("read_from_file: Problem reading file ", filepath)
+        return -1
+    end
+    
+    width  = read(io, Int32)
+    height = read(io, Int32)
+    
+    flow = Array{Array{Float64 ,1}, 2}(undef, height, width)
+    for i in 1:height
+        for j in 1:width
+            if eof(io) == false
+                fx = read(io, Float32)
+                fy = read(io, Float32)
+                flow[i, j] = [fx, fy]
+            end
+        end
+    end
+    close(io)
+    return convert(Array{SArray{Tuple{2},Float64,1,2},2}, flow)
+end
+
+
+"""
+```
+evaluate_flow_error(ground_truth_flow, estimated_flow, EndpointError())
+```
+
+Returns a 2-Dimensional array that matches the dimensions of the input flow vector and
+that depicts the end point error between the estimated flow and the ground truth flow.
+
+# Details
+If the estimated flow at a point is (u0, v0) and ground truth flow is (u1, v1), then 
+error will be sqrt[(u0 - u1)^2 + (v0 - v1)^2] at that point.
+
+# Arguments
+The flow parameters needs to be two-dimensional arrays of length-2 vectors (of type SVector) 
+which represent the displacement of each pixel.
+
+# Example
+
+Compute the end point error between two flows.
+```julia
+
+using ImageTracking
+
+result = evaluate_flow_error(ground_truth_flow, estimated_flow, EndpointError())
+
+imshow(result)
+```
+
+# References
+[1] S. Baker, D. Scharstein, JP Lewis, S. Roth, M.J. Black, and R. Szeliski. A database and
+evaluation methodology for optical flow.International Journal of Computer Vision, 92(1):1–31, 2011.
+"""
+function evaluate_flow_error(ground_truth_flow::Array{SVector{2, Float64}, 2}, estimated_flow::Array{SVector{2, Float64}, 2}, error_type::EndpointError)
+    result = Array{Float64, 2}(undef, size(estimated_flow))
+    for i in 1:size(estimated_flow)[1]
```

Comment:
Might be cleaner to use `CartesianIndices` here.

```julia
for i in CartesianIndices(estimated_flow)
            p1 = estimated_flow[i]
            p2 = ground_truth_flow[i] 
            # Etc.
            result[i] = ...
            # Etc.
end
```

Generally, you want to avoid expression like ``1:size(estimated_flow)[2]`. The brackets required to index into the result of size look particularly jarring  in this context my opinion.

Reply:
Done